packages/prismic/src/helpers/index.ts

```diff
@@ -42,44 +42,66 @@ export const getPageSlugs = (page: Document): string[] => page.slugs;
 
 export const getPageLang = (page: Document): string => page.lang;
 
-export const getBlocks = ({ data }: Document, blockName?: string): string | string[] => {
-  const blockKeys = Object.keys(data || {})
+type ElementNameType = string | string[] | null | undefined;
+type FilterSlice = (slice: PrismicSlice) => boolean;
+
+export const getBlocks = (data: any, blockName?: ElementNameType, transform: TransformBlock = transformFunction): string | string[] => {
+  if (typeof data !== 'object' || data === null) {
+    return blockName ? '' : [];
+  }
+
+  const blockKeys = Object.keys(data)
     .filter((key) => key !== 'body')
     .filter((key) => data[key] !== null);
 
-  if (blockName) {
+  if (typeof blockName === 'string') {
     const key = blockKeys.find((blockKey) => blockKey === blockName);
 
     if (key === undefined) {
       return '';
     }
 
-    return transformBlock(data[key]);
+    return transform(data[key]);
   }
 
-  return blockKeys.map((key) => transformBlock(data[key]));
-};
+  const filteredBlockKeys = Array.isArray(blockName)
+    ? blockKeys.filter((key) => blockName.includes(key))
+    : blockKeys;
 
-export const getSlices = ({ data }: Document, sliceType?: string): string[] | Array<string[]> => {
-  const renderSliceElements = (slice: PrismicBlock): string => Object.keys(slice)
-    .filter((key) => slice[key] !== null && Object.keys(slice[key].length !== 0))
-    .map((key) => transformBlock(slice[key]))
-    .join('');
+  return filteredBlockKeys.map((key) => transform(data[key]));
+};
 
+export const getSlices = ({ data }: Document, sliceType?: ElementNameType | FilterSlice): any => {
   const slices = data.body as PrismicSlice[];
 
-  if (sliceType) {
-    const foundSlice = slices.find((slice) => slice.slice_type === sliceType);
+  if (typeof sliceType === 'string') {
+    const foundSlices = slices.find((slice) => slice.slice_type === sliceType);
 
-    if (!foundSlice) {
-      return [];
-    }
+    const { primary, items } = foundSlices || { primary: null,
+      items: [] };
+
+    return {
+      primary,
+      items
+    };
+  }
 
-    return [foundSlice.primary || {}, ...foundSlice.items].map((item) => renderSliceElements(item));
+  const filteredSlices = Array.isArray(sliceType)
+    ? slices.filter((slice) => sliceType.includes(slice.slice_type))
+    : slices;
+
+  if (typeof sliceType === 'function') {
+    return filteredSlices
```

Comment:
maybe just make it a `let` and overwrite here to have single return e.g.
```suggestion
    filteredSlices =  filteredSlices.filter((slice) => sliceType(slice));
```